File: iOS/Great-Vinyl-Exchange/Great-Vinyl-Exchange/Models/Record.swift
//
//  Record.swift
//  Great-Vinyl-Exchange
//
//  Created by Daniel Ley on 11/20/16.
//  Copyright © 2016 daniel. All rights reserved.
//

import Foundation
import UIKit
import Firebase

class Record: NSObject {
    
    // MARK: Properties
    var image: UIImage?
    var sender: User
    var reciever: User
    var dateRecieved: String
    var monthAssigned: String
    var yearAssigned: String
    
    var artist: String?
    var album: String?
    var name: String?
    
    //Incorporate Discogs Info at somepoint
    
    // MARK: Initialization
    init?(image: UIImage, sender: User, reciever: User, month: String, year: String) {
        self.image = image
        self.sender = sender
        self.reciever = reciever
        self.monthAssigned = month
        self.yearAssigned = year
        
        let currentDate : NSDate = NSDate()
        let dateFromatter = DateFormatter()
        dateFromatter.dateStyle = DateFormatter.Style.short
        let convertedDate = dateFromatter.string(from: currentDate as Date)
        self.dateRecieved = convertedDate
    }
    
    
    func toAnyObject() -> AnyObject {
        let imageData: NSData = UIImagePNGRepresentation(image!)! as NSData
        
        // TODO: incorporate adding images to Firebase.
        // let base64String = imageData.base64EncodedString(options: NSData.Base64EncodingOptions.Encoding64CharacterLineLength)
        var record = [String:String]()
        record = [
            "image": "",
            "sender": sender.name,
            "reciever": reciever.name,
            "dateRecieved": dateRecieved,
            "monthAssigned": monthAssigned,
            "yearAssigned": yearAssigned
        ]
        return record as AnyObject
    }
}

File: Android/GreatVinylExchange/app/src/main/java/Models/Record.java
package Models;

import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

/**
 * Created by DanielLey on 11/27/16.
 */
public class Record {
    // MARK: Properties
    User sender;
    User reciever;
    String month;
    String year;
    String dateRecieved;
    // Need Image

    // MARK: Constructor
    public Record(User xSend, User xRec, String xMonth, String xYear) {
        sender = xSend;
        reciever = xRec;
        month = xMonth;
        year = xYear;

        DateFormat df = new SimpleDateFormat("dd/MM/yy");
        Date dateobj = new Date();
        dateRecieved = df.format(dateobj);
    }

    // MARK: Object Description Method
    private Map toAnyObject() {
        Map<String, String> record = new HashMap<String, String>();
        record.put("Sender",sender.name);
        record.put("Reciever",reciever.name);
        record.put("Month",month);
        record.put("Year",year);
        record.put("DateRecieved",dateRecieved);
        return record;
    }
}
